### PR TITLE
RIG construction by Robotic

### PR DESCRIPTION
**Pull request by Sindorman**
This PR lets Robotics lab to be able to assemble different RIGS. Process is similar to Mechs construction.

Rig that will be allowed to be constructed: EVA, industrial, medical, CE, AMI, light, hazard, combat, cybersuit(illegal tech)

Part of #6032 update

Misc changes:

- Moved mechfab_designs.dm to designs folder.

- Decreases bullet armour for industrial RIG

- Gives stealth RIG slightly better armour values than regular light RIG

- Slightly increases hazard RIG bullet and radiation protection.

TODO:

- [x] Make generic framework for RIG construction based of Mech construction.

- [x] Reduce Mech's code to be generic and require less lines

- [x] Define all RIGS

- [x] Define all RIGs circuits

- [x] Add all RIGS that can be constructed to blueprints and assign their material and tech levels.

- [x] Get sprites from Kyres and update them in code.

### CODE REVIEW

File: code/game/objects/items/weapons/circuitboards/rig.dm [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	origin_tech = list(TECH_DATA = 4)
+
+/obj/item/weapon/circuitboard/rig_assembly/civilian/industrial
+	name = "industrial suit central circuit board"
+
+/obj/item/weapon/circuitboard/rig_assembly/civilian/eva
+	name = "EVA suit central circuit board"
+
+/obj/item/weapon/circuitboard/rig_assembly/civilian/ce
+	name = "an advanced void suit central circuit board"
```
Comment (Alberyk) [inline code/game/objects/items/weapons/circuitboards/rig.dm]: "advanced void suit central circuit board" sounds better here.

File: code/modules/clothing/spacesuits/rig/rig_construction.dm [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+					//6
+					list("key"=/obj/item/weapon/wirecutters,
+							"backkey"=/obj/item/weapon/screwdriver,
+							"desc"="The wiring is added"),
+					//7
+					list("key"=/obj/item/stack/cable_coil,
+							"desc"="The wiring is removed added."),
+					)
+
+	action(atom/used_atom,mob/user as mob)
```
Comment (Alberyk) [inline code/modules/clothing/spacesuits/rig/rig_construction.dm]: Relative pathing here.

File: code/modules/clothing/spacesuits/rig/rig_construction.dm [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+		return check_step(used_atom,user)
+
+	custom_action(index, diff, atom/used_atom, mob/user)
```
Comment (Alberyk) [inline code/modules/clothing/spacesuits/rig/rig_construction.dm]: Relative pathing here.

File: code/modules/clothing/spacesuits/rig/rig_construction.dm [hunk 4]
```diff
@@ -0,0 +1,432 @@
...
+					//13
+					list("key"=/obj/item/weapon/wirecutters,
+							"backkey"=/obj/item/weapon/screwdriver,
+							"desc"="The wiring is added"),
+					//14
+					list("key"=/obj/item/stack/cable_coil,
+							"desc"="The wiring is removed added."),
+					)
+
+	action(atom/used_atom,mob/user as mob)
```
Comment (Alberyk) [inline code/modules/clothing/spacesuits/rig/rig_construction.dm]: Relative pathing here.

File: code/modules/clothing/spacesuits/rig/rig_construction.dm [hunk 5]
```diff
@@ -0,0 +1,432 @@
...
+		return check_step(used_atom,user)
+
+	custom_action(index, diff, atom/used_atom, mob/user)
```
Comment (Alberyk) [inline code/modules/clothing/spacesuits/rig/rig_construction.dm]: Relative pathing here.

File: code/modules/research/designs/circuit_designs.dm [hunk 6]
```diff
@@ -529,4 +529,95 @@
...
+
+/datum/design/circuit/eva
+	name = "EVA suit central circuit board"
+	id = "eva_rig"
+	req_tech = list(TECH_DATA = 3)
+	build_path = /obj/item/weapon/circuitboard/rig_assembly/civilian/eva
+	sort_string = "WAAAG"
+
+/datum/design/circuit/ce
+	name = "an advanced void suit central circuit board"
```
Comment (Alberyk) [inline code/modules/research/designs/circuit_designs.dm]: "advanced void suit central circuit board"

File: code/modules/clothing/spacesuits/rig/rig_construction.dm [hunk 7]
```diff
@@ -0,0 +1,432 @@
...
+					)
+
+/datum/construction/reversible/rig_assembly/civilian/action(atom/used_atom,mob/user as mob)
+	return check_step(used_atom,user)
+
+/datum/construction/reversible/rig_assembly/civilian/custom_action(index, diff, atom/used_atom, mob/user)
+	if(!..())
+		return 0
+
+	var/obj/item/rig_assembly/r
```
Comment (skull132) [inline code/modules/clothing/spacesuits/rig/rig_construction.dm]: This entire deal can be shortened:
```dm
	var/obj/item/rig_assembly/r = holder
	if (!istype(r))
		return 0
```

File: code/modules/clothing/spacesuits/rig/rig_construction.dm [hunk 8]
```diff
@@ -0,0 +1,432 @@
...
+					)
+
+/datum/construction/reversible/rig_assembly/combat/action(atom/used_atom,mob/user as mob)
+	return check_step(used_atom,user)
+
+/datum/construction/reversible/rig_assembly/combat/custom_action(index, diff, atom/used_atom, mob/user)
+	if(!..())
+		return 0
+
+	var/obj/item/rig_assembly/r
```
Comment (skull132) [inline code/modules/clothing/spacesuits/rig/rig_construction.dm]: Same comment here as above.

File: code/modules/research/designs/mechfab_designs.dm [hunk 9]
```diff
@@ -862,4 +862,79 @@
...
+
+/datum/design/rig/hazard
+	name = "hazard hardsuit control module"
+	desc = "An assembly for a security hardsuit designed for prolonged EVA in dangerous environments."
+	id = "rig_hazard"
+	build_path = /obj/item/rig_assembly/combat/hazard
+	req_tech = list(TECH_MATERIAL = 4, TECH_ENGINEERING = 3, TECH_MAGNET = 3, TECH_POWER = 3, TECH_COMBAT = 3)
+	materials = list(DEFAULT_WALL_MATERIAL = 45000, "glass" = 20000)
+
+/datum/design/rig/combat
```
Comment (VTCobaltblood) [inline code/modules/research/designs/mechfab_designs.dm]: Why are these legal? They have:
```
	initial_modules = list(
		/obj/item/rig_module/mounted,
		/obj/item/rig_module/vision/thermal,
		/obj/item/rig_module/grenade_launcher,
		/obj/item/rig_module/ai_container,
		/obj/item/rig_module/power_sink,
		/obj/item/rig_module/electrowarfare_suite,
		/obj/item/rig_module/chem_dispenser/combat
		)
```
Aren't they incredibly OP to be produced by the station?

File: code/modules/clothing/spacesuits/rig/rig_construction.dm [hunk 10]
```diff
@@ -0,0 +1,428 @@
...
+				list("key"=null,
+						"backkey"=/obj/item/weapon/wirecutters,
+						"desc"="The wiring is adjusted"),
+				//6
+				list("key"=/obj/item/weapon/wirecutters,
+						"backkey"=/obj/item/weapon/screwdriver,
+						"desc"="The wiring is added"),
+				//7
+				list("key"=/obj/item/stack/cable_coil,
+						"desc"="The wiring is removed added."),
```
Comment (VTCobaltblood) [inline code/modules/clothing/spacesuits/rig/rig_construction.dm]: what

File: code/modules/clothing/spacesuits/rig/rig_construction.dm [hunk 11]
```diff
@@ -0,0 +1,428 @@
...
+						"backkey"=/obj/item/weapon/crowbar,
+						"desc"="Central control module is installed"),
+				//5
+				list("key"=null,
+						"backkey"=/obj/item/weapon/wirecutters,
+						"desc"="The wiring is adjusted"),
+				//6
+				list("key"=/obj/item/weapon/wirecutters,
+						"backkey"=/obj/item/weapon/screwdriver,
+						"desc"="The wiring is added"),
```
Comment (VTCobaltblood) [inline code/modules/clothing/spacesuits/rig/rig_construction.dm]: with a screwdriver?

File: code/modules/research/designs/mechfab_designs.dm (hunk 9 above)
Comment (Sindorman) [reply code/modules/research/designs/mechfab_designs.dm]: Uhm, that is for equipped version only. RND produces vanilla, equipped version is only use for adminbus spawn. Also if you noticed I nerfed armour values to more appropriate level where it is not as good as say ERT or Mercs RIGs

File: code/modules/clothing/spacesuits/rig/rig_construction.dm (hunk 11 above)
Comment (Sindorman) [reply code/modules/clothing/spacesuits/rig/rig_construction.dm]: Yes, that is to go back to step of removing wire. Unfortunately I cannot make it wirecutters as to proceed to the next step you also need wirecutters.

**Pull request merged**